distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json/Extensions.enso

```diff
@@ -52,7 +52,7 @@ Number.to_js_object self = case self of
         ## JS Safe Integer range -(2^53 - 1) to (2^53 - 1)
         js_max_integer = 9007199254740991
         if self >= -js_max_integer && self < js_max_integer then self else
-            JS_Object.from_pairs [["type", "BigInt"], ["value", self.to_text]]
+            JS_Object.from_pairs [["type", "BigInt"], ["value", self.to_text], ["_display_text_", self.to_display_text]]
```

Comment:
Not sure if something to tackle in this PR, but looking at this I feel like we may be making an architecture mistake and I'd like to discuss at least in terms of possible follow-up.

We are including "display" (view) data in `to_js_object` which to me sounds more like a "serialization" thing. I assume I will want to use `to_js_object` when writing a Table or any other object to a `.json` file, right? And in such a serialized format, I really doubt I want a `_display_text_` field - at least by default.

I think the issue is `to_js_object` is trying to do too many things at the same time. The JSON representation for _serialization_ and for _view_ are just two different concepts and mixing them sounds like not the best idea.

What if instead, we kept `to_js_object` as is and relied on a _separate_ method for the visualization? It can either be `to_default_visualization_data` or some new `to_js_object_for_display` or something.

What do you think?

Reply:
Have removed this from the two places outside of the Table Viz.